docs/user/modules/python/pages/pyfeelppmor/reducedbasis.adoc

```diff
@@ -4,112 +4,355 @@ include::user:ROOT:partial$header-macros.adoc[]
 
-model.setOnlineAssembleDEIM(assembleOnlineDEIM)
-
-heatBoxMDEIM=heat(dim=DIM,order=1)
-meshMDEIM = model.getMDEIMReducedMesh()
-heatBoxMDEIM.setMesh(meshMDEIM)
-heatBoxMDEIM.init()
-
 def assembleOnlineMDEIM(mu):
     for i in range(0,mu.size()):
         heatBoxMDEIM.addParameterInModelProperties(mu.parameterName(i),mu(i))
     heatBoxMDEIM.updateParameterValues()
     return heatBoxMDEIM.assembleMatrix()
 
+model.setOnlineAssembleDEIM(assembleOnlineDEIM)
 model.setOnlineAssembleMDEIM(assembleOnlineMDEIM)
-----
 
-Then finalize the initialization :
-
-[soucre,python]
-----
 model.postInitModel()
 model.setInitialized(True)
 ----
 
+== Offline generation
 
-== Access to a parameter
+Information about the parameter space stem:[D^\mu]
 
-[source,python]
+[source, python]
 ----
 Dmu = model.parameterSpace()
-mu = Dmu.element(True, False)
+print(Dmu.parameterNames())
+print(Dmu.mumin())
+print(Dmu.mumax())
+----
+
+[%collapsible.result]
+.Results
+====
+----
+['Bi'    ,'k_1'   ,'k_2'   ,'k_3'   ,'k_4']
+[1.00e-02,1.00e-01,1.00e-01,1.00e-01,1.00e-01]
+[1.00e+00,1.00e+01,1.00e+01,1.00e+01,1.00e+01]
+----
+====
+
+[source, python]
+----
+mubar = Dmu.element(True, False)
+mubar.setParameters(default_parameter)
+print("mubar =")
+mubar.view()
+----
+
+[%collapsible.result]
+.Results
+====
+----
+mubar =
+Bi	0.01
+k_1	0.1
+k_2	0.1
+k_3	0.1
+k_4	0.1
+----
+====
+
+Get affine decomposition stem:[A(\mu) = \displaystyle\sum_q \beta^q_A(\mu) A^q], stem:[F(\mu)=\displaystyle\sum_p \beta_F^p(\mu) F^(\mu)].
+To use the matrices and vector in the class, we need to convert themn to `petsc4py` objects, using the functions `convertToPetscMat` and `convertToPetscVec`.
+
+[source, python]
+----
+affineDecomposition = model.getAffineDecomposition()
+Aq_ = affineDecomposition[0]
+Fq_ = affineDecomposition[1]
+
+Aq = mor_rb.convertToPetscMat(Aq_[0])
+Fq = []
+for f in Fq_:
+    Fq.append(mor_rb.convertToPetscVec(f[0]))
+
+print("Aq", Aq)
+print("Fq", Fq)
+----
+
+[%collapsible.result]
+.Results
+====
+----
+Aq [<petsc4py.PETSc.Mat object at 0x7f5f693abba0>, <petsc4py.PETSc.Mat object at 0x7f5f7000d210>, <petsc4py.PETSc.Mat object at 0x7f5f693fd3f0>, <petsc4py.PETSc.Mat object at 0x7f5f693fd3a0>, <petsc4py.PETSc.Mat object at 0x7f5f693fd210>, <petsc4py.PETSc.Mat object at 0x7f5f693fd260>]
+Fq [[<petsc4py.PETSc.Vec object at 0x7f5f693fd350>], [<petsc4py.PETSc.Vec object at 0x7f5f693fd2b0>], [<petsc4py.PETSc.Vec object at 0x7f5f693fd1c0>]]
+----
+====
+
+Now we create the object `reduced_basis`.
+
+[source, python]
+----
+rb = mor_rb.reducedbasisOffline(Aq=Aq, Fq=Fq, model=model, mubar=mubar,
+                output_names=output_names, use_dual_norm=False)
+----
+
+[%collapsible.result]
+.Results
+====
+----
+[reducedbasis] Online rb initialized
+[slepc4py] number of (smaller) eigenvalues computed : 16
+[reducedbasis] Constant of coercivity : 0.9999999999999977
+[slepc4py] number of eigenvalues computed : 16
+[reducedbasis] Constant of continuity : 1.0000000000000018
```

Comment:
continuity and coercivity are the same ? does it mean that the parameter for use is the one for the scalar product ?

Reply:
make a note for this